cmd/agent/common/misconfig/global.go

```diff
@@ -8,19 +8,35 @@ package misconfig
 import "github.com/DataDog/datadog-agent/pkg/util/log"
 
 // ToLog outputs warnings about common misconfigurations in the logs
-func ToLog() {
+func ToLog(agent AgentType) {
 	for name, check := range checks {
-		if err := check(); err != nil {
-			log.Warnf("misconfig: %s: %v", name, err)
+		if _, ok := check.supportedAgents[agent]; ok {
+			if err := check.run(); err != nil {
+				log.Warnf("misconfig: %s: %v", name, err)
+			}
 		}
 	}
 }
 
+// AgentType denotes the type of agent that is currently being run
+type AgentType int
 type checkFn func() error
+type check struct {
+	name            string
+	run             checkFn
+	supportedAgents map[AgentType]struct{}
+}
+
+const (
+	// CoreAgent is an enum to denote AgentType
+	CoreAgent = iota
+	// ProcessAgent is an enum to denote AgentType
```

Comment:
`CoreAgent` and `ProcessAgent` belongs to the same enum, `AgentType`. We should probably update the comments to reflect this.

Reply:
Thanks for the suggestion! I updated the type signature to be `AgentType`